libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1242,25 +1243,27 @@ string ABIFunctions::abiDecodingFunctionCalldataArray(ArrayType const& _type)
 					arrayPos := add(offset, 0x20)
 					if gt(add(arrayPos, mul(length, <stride>)), end) { <revertStringPos> }
 				}
-			)";
+			)");
+			w("revertStringOffset", revertReasonIfDebug("ABI decoding: invalid calldata array offset"));
+			w("revertStringLength", revertReasonIfDebug("ABI decoding: invalid calldata array length"));
+		}
 		else
-			templ = R"(
+		{
+			w = Whiskers(R"(
 				// <readableTypeName>
 				function <functionName>(offset, end) -> arrayPos {
 					arrayPos := offset
 					if gt(add(arrayPos, mul(<length>, <stride>)), end) { <revertStringPos> }
 				}
-			)";
-		Whiskers w{templ};
-		// TODO add test
```

Comment:
Don't we actually have these tested now?

Reply:
No, we don't.